modules/files.nix

```diff
@@ -54,9 +64,40 @@ let
             # This means our `source` definition has the same priority as `text`.
             lib.mkDerivedConfig options.text (pkgs.writeText derivationName)
           );
+          finalSource =
+            # Byte compile lua files if performance.byteCompileLua option is enabled
+            if
+              lib.hasSuffix ".lua" config.target
+              && topConfig.performance.byteCompileLua.enable
+              && topConfig.performance.byteCompileLua.configs
+            then
+              if lib.isDerivation config.source then
+                # Source is a derivation, add byteCompileLuaHook to it
+                (config.source.overrideAttrs (
+                  prev:
+                  {
+                    nativeBuildInputs = prev.nativeBuildInputs or [ ] ++ [ helpers.byteCompileLuaHook ];
+                  }
+                  // lib.optionalAttrs (prev ? buildCommand) {
+                    buildCommand = ''
+                      ${prev.buildCommand}
+                      runHook postFixup
+                    '';
+                  }
```

Comment:
Exposing this function in lib/builders.nix could be useful, as it provides an easy way to byte-compile selected plugins if required. It should also allow to re-use the same function here & in overriding the plugins, reducing the risk somebody else forgets to modify one if need arises.